src/library_tty.js

```diff
@@ -153,8 +163,13 @@ mergeInto(LibraryManager.library, {
         if (val === null || val === {{{ charCode('\n') }}}) {
           Module['print'](UTF8ArrayToString(tty.output, 0));
           tty.output = [];
-        } else {
-          if (val != 0) tty.output.push(val); // val == 0 would cut text output off in the middle.
+        } else if (val != 0) {
```

Comment:
This change looks like intent to just micro-optimize away characters, in that case, I think we can even do `} else if (val) {` and drop the `val != 0`, although either way is just fine.